base/applications/rapps/geninst.cpp

```diff
@@ -835,3 +840,46 @@ UninstallGenerated(CInstalledApplicationInfo &AppInfo, UninstallCommandFlags Fla
     g_pInfo = &Info;
     return CreateUI(Info.Silent, UninstallThread) ? !Info.Error : FALSE;
 }
+
+HRESULT
+ExtractArchiveForExecution(PCWSTR pszArchive, const CStringW &PackageName, CStringW &TempDir, CStringW &App)
+{
+    WCHAR buf[MAX_PATH], buf2[MAX_PATH];
```

Comment:
Please use self-documented names here. What's "buf" used for? and "buf2"?
I would suggest you may use CStringW if possible (it's possible with buf2, because the wsprintfW call below would become: `buf2.Format(...)` ).